scipy/stats/tests/test_stats.py

```diff
@@ -7674,6 +7674,28 @@ def test_inf_values(self):
                 stats.wasserstein_distance([1, 2, np.inf], [np.inf, 1]),
                 np.nan)
 
+    @pytest.mark.parametrize('u_size', [1, 10, 300])
+    @pytest.mark.parametrize('v_size', [1, 10, 300])
+    def test_optimization_vs_analytical(self, u_size, v_size):
+        rng = np.random.default_rng(45634745675)
+        # Test when u_weights = None, v_weights = None
+        u_values = rng.random(size=(u_size, 1))
+        v_values = rng.random(size=(v_size, 1))
+        u_values_flat = u_values.ravel()
+        v_values_flat = v_values.ravel()
+        # These two calculations are done using different backends
+        # but they must be equal
+        assert_allclose(stats.wasserstein_distance(u_values, v_values),
+                        stats.wasserstein_distance(u_values_flat, v_values_flat))
+        # Test with u_weights and v_weights specified.
+        u_weights = rng.random(size=u_size)
+        v_weights = rng.random(size=v_size)
+        d1 = stats.wasserstein_distance(u_values, v_values,
+                                        u_weights, v_weights)
+        d2 = stats.wasserstein_distance(u_values_flat, v_values_flat,
+                                        u_weights, v_weights))
```

Comment:
```suggestion
                                        u_weights, v_weights)
```

Reply:
I am sorry. I couldn't commit your suggestion, because I saw it after I made this exact fix.